src/Mailer/Message/CitizenActionNotificationMessage.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace AppBundle\Mailer\Message;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\CitizenAction;
+use Ramsey\Uuid\Uuid;
+
+class CitizenActionNotificationMessage extends Message
+{
+    public static function create(
+        array $recipients,
+        Adherent $host,
+        CitizenAction $citizenAction,
+        string $citizenActionAttendLink
+    ): self {
+        if (!$recipients) {
+            throw new \InvalidArgumentException('At least one recipient is required.');
+        }
+
+        $recipient = array_shift($recipients);
+        if (!$recipient instanceof Adherent) {
+            throw new \InvalidArgumentException(sprintf(
+                'This message builder requires a collection of %s instances',
+                Adherent::class
+            ));
+        }
+
+        $message = new static(
+            Uuid::uuid4(),
+            '326404',
+            $recipient->getEmailAddress(),
```

Comment:
I know it's not you but it's a real pain not to manage one or multiple recipient directly in `Message`. I spend some time trying to understand while it should be straightforward.

One day we should fix that :-)

Reply:
Clearly a thing that could be done in addition of #2228